packages/search/lib/pullElasticsearch.js

```diff
@@ -112,8 +119,10 @@ module.exports = async ({
         redis
       })
 
-      debug('waiting grace period', { writeAlias, index })
-      await timeout(1000 * 5)
+      if (doWait) {
```

Comment:
Lines 107 - 110 look like copy paste. Could you generalize it?

Something like `await inserts.dict[name].after({}).then(logAndReturn)`

At the same time, you could choose a function name that might suggest why this is sometimes needed? Or we could pass a variable with a name that tells us why we need to wait, instead of what we do?

e.g. `foo({ensurePropagation: true})` statt `foo({doWait: true})`?

Reply:
I'd rather not refactor the 2 copy/pasted lines. The style of logging in this file could be better, but that's out of scope for my PR.